### PR TITLE
Refactor/standardize skill id usage

### PR DESCRIPTION
**Pull request by JarbasAl**
relates to https://github.com/MycroftAI/mycroft-core/issues/2891
depends on #27 and #29

This commit changes every usage of skill name to the deterministic skill_id

There are several references in the code base to MycroftSkill self.name, this is not deterministic, by default it takes the name of the class but can be defined by skill developers and change in each skill update.

self.name/self.skill_id are used in the __init__ method, but skill_id is not ensured to be defined at this step, potentially introducing issues depending on when certain methods are called, this is particularly troublesome when skills use the __init__ method since behavior may change depending on when super() is called

also changes all instances in MycroftSkill to use f strings instead of string.format() syntax

### CODE REVIEW

File: mycroft/skills/event_scheduler.py [hunk 1]
```diff
@@ -281,20 +280,35 @@ def shutdown(self):
...
+        """Attach the skill_id of the parent skill
+
+        Args:
+            sched_id (str): skill_id of the parent skill
+        """
+        # NOTE: can not rename sched_id kwarg to keep api compatibility
+        self.skill_id = sched_id
 
     def _create_unique_name(self, name):
         """Return a name unique to this skill using the format
```
Comment (NeonDaniel) [inline mycroft/skills/event_scheduler.py:314]: Maybe clarify that this is a unique `schedule name` so it's tied to the scheduler and not the skill manager?

File: mycroft/skills/mycroft_skill/mycroft_skill.py [hunk 2]
```diff
@@ -133,7 +135,7 @@ def __init__(self, name=None, bus=None, use_settings=True):
         self.gui = SkillGUI(self)
 
         self._bus = None
-        self._enclosure = None
+        self._enclosure = EnclosureAPI()
```
Comment (NeonDaniel) [inline mycroft/skills/mycroft_skill/mycroft_skill.py:138]: This change seems out of scope for this PR

File: mycroft/skills/mycroft_skill/mycroft_skill.py [hunk 3]
```diff
@@ -215,6 +216,10 @@ def _startup(self, bus, skill_id=""):
             self.skill_id = skill_id or basename(self.root_dir)
             self.intent_service.set_id(self.skill_id)
             self.event_scheduler.set_id(self.skill_id)
+            self.enclosure.set_id(self.skill_id)
+
+            # initialize anything that depends on skill_id
+            self.log = LOG.create_logger(self.skill_id)
```
Comment (NeonDaniel) [inline mycroft/skills/mycroft_skill/mycroft_skill.py:222]: Is it desirable to create a new logger instead of using LOG directly? I know this is how Mycroft did it, but I never really understood why

File: mycroft/skills/mycroft_skill/mycroft_skill.py [hunk 4]
```diff
@@ -287,6 +292,28 @@ def enclosure(self):
...
+        else:
+            LOG.error('Skill not fully initialized. Move code ' +
+                      'from  __init__() to initialize() to correct this.')
+            LOG.error(simple_trace(traceback.format_stack()))
+            raise Exception('Accessed MycroftSkill.file_system in __init__')
+
+    @file_system.setter
+    def file_system(self, fs):
+        """Provided mainly for backwards compatibility with derivative MycroftSkill classes
+        Skills are advised against redefining the file system directory"""
```
Comment (NeonDaniel) [inline mycroft/skills/mycroft_skill/mycroft_skill.py:314]: Maybe log a warning if this is intended to be deprecated?

**Review by NeonDaniel — CHANGES_REQUESTED** [review]
Some of the MycroftSkill init changes feel unrelated to the skill_id changes and I think are more than a refactoring since they change how `init` behaves. Also left some comments

File: mycroft/skills/mycroft_skill/mycroft_skill.py (hunk 2 above)
Comment (JarbasAl) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:138]: enclosure was created in self.bind with bus and skill id, but that is not needed.

with this PR we use set_bus and set_id like in other objects for consistency. so its as good a place as any to initialize the object

File: mycroft/skills/mycroft_skill/mycroft_skill.py (hunk 3 above)
Comment (JarbasAl) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:222]: changes the name of the logger internally and help filtering logs

File: mycroft/skills/mycroft_skill/mycroft_skill.py (hunk 4 above)
Comment (JarbasAl) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:314]: its not deprecated but it should be considered read only

i made it a property but previously it was a variable, i allow assigning to it to keep the api compatible in case something like neon is overriding it

File: mycroft/skills/event_scheduler.py (hunk 1 above)
Comment (JarbasAl) [reply mycroft/skills/event_scheduler.py:314]: this is the skill_id we are all used to, comes from the parent skill. no idea why sched_id was used but `schedule name` is not a thing

File: mycroft/skills/mycroft_skill/mycroft_skill.py (hunk 4 above)
Comment (JarbasAl) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:314]: the reason this changed is that its easier to account for skill_id this way, otherwise i would need some _init_filesystem method, but this can cause issues depending on what a skill does between init and initialize

i think homescreen skill runs (used to?) into this when handling wallpapers, it accesses file system in init,  so doing it this way indicates that breaking change

Comment (NeonDaniel) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:314]: Is the setter going to remain, or is it planned to deprecate it later? If this is sticking around then this is fine

Comment (JarbasAl) [reply mycroft/skills/mycroft_skill/mycroft_skill.py:314]: it will stay, not planned for removal

**Review by NeonDaniel — APPROVED** [review]
